angularSTEP/src/app/upload-recipe/upload-recipe.component.ts

```diff
@@ -76,44 +108,55 @@ export class UploadRecipeComponent {
     return formattedName;
   }
 
-  removeNumsAndSymbolsAtStart(originalString: string): string | undefined {
-      var startIndex: number = -1;
+  removeNumsAndSymbolsAtStart(originalString: string): string {
```

Comment:
Yay! Sticking to a single non-undefined return type is definitely better than letting undefineds spread around  :)